src/fields/field_algo_base.impl.hh

```diff
@@ -53,31 +54,24 @@ string FieldAlgorithmBase<spacedim, Value>::template_name() {
 
 
 template <int spacedim, class Value>
-Input::Type::AbstractRecord & FieldAlgorithmBase<spacedim, Value>::get_input_type(const typename Value::ElementInputType *element_input_type) {
-	const it::Selection *element_input_sel = nullptr;
-	if (element_input_type != nullptr) {
-		const it::Selection * sel_type = dynamic_cast<const it::Selection *>(element_input_type);
-		if (sel_type != NULL ) element_input_sel = sel_type;
-	}
+Input::Type::AbstractRecord & FieldAlgorithmBase<spacedim, Value>::get_input_type() {
+	return it::AbstractRecord("Field:"+template_name(), "Abstract record for all time-space functions.")
```

Comment:
- AbstractRecord::content_hash() should has also attribute "parameters". Then we can set template_name() to contain just specification of dimension.